flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/strategy/SchedulingTopology.java

```diff
@@ -50,4 +52,11 @@
 	 * @return Optional containing the respective scheduling result partition or none if the partition does not exist
 	 */
 	Optional<SchedulingResultPartition> getResultPartition(IntermediateResultPartitionID intermediateResultPartitionId);
+
+	/**
+	 * Looks up the {@link InputDependencyConstraint} for the given {@link JobVertexID}.
+	 * @param jobVertexId identifying the respective job vertex id
+	 * @return Optional containing the respective input dependency constraint or none if the job vertex does not exist
+	 */
+	Optional<InputDependencyConstraint> getInputDependencyConstraint(JobVertexID jobVertexId);
```

Comment:
Since `InputDependencyConstraint` is a property of the `JobVertex`, I think it makes more sense to add this to the `SchedulingExecutionVertex` (without the `Optional`)

Reply:
OK